### ISSUE TITLE
[Bug] Server-side Lexical to HTML serializer is not working for nested fields

### ISSUE THREAD

**Issue by hunghvu** [payloadcms/payload#4034]
### Describe the Bug

At the moment, the `lexicalHtml` does not work on field types that have children fields. Those fields are `group`, `array`, `collapsible`, `row`, and `tabs`. If we extrapolate the issue, there can be unknown similar-in-nature nested fields that break the functionality of `lexicalHtml`.

Related issues: #4023 

### To Reproduce

In the code below, I went over some settings and commented the behavior of `lexicalHtml`. In most cases, the error is 

```
cms:dev: [04:30:06] ERROR (payload): Error: You cannot use the lexicalHTML field because the lexical field was not found
cms:dev:     at currentHook (/home/dev/Documents/github/hungvu.tech/node_modules/.pnpm/@payloadcms+richtext-lexical@0.1.16_@lexical+clipboard@0.12.2_payload@2.0.15_typescript@5.2.2_yjs@13.6.8/node_modules/@payloadcms/richtext-lexical/src/field/features/converters/html/field/index.ts:42:19)
cms:dev:     at /home/dev/Documents/github/hungvu.tech/node_modules/.pnpm/payload@2.0.15_typescript@5.2.2_webpack@5.89.0/node_modules/payload/src/fields/hooks/afterRead/promise.ts:222:37
cms:dev:     at processTicksAndRejections (node:internal/process/task_queues:95:5)
cms:dev:     at async promise (/home/dev/Documents/github/hungvu.tech/node_modules/.pnpm/payload@2.0.15_typescript@5.2.2_webpack@5.89.0/node_modules/payload/src/fields/hooks/afterRead/promise.ts:190:7)
cms:dev:     at async Promise.all (index 11)
cms:dev:     at async afterRead (/home/dev/Documents/github/hungvu.tech/node_modules/.pnpm/payload@2.0.15_typescript@5.2.2_webpack@5.89.0/node_modules/payload/src/fields/hooks/afterRead/index.ts:67:3)
cms:dev:     at async create (/home/dev/Documents/github/hungvu.tech/node_modules/.pnpm/payload@2.0.15_typescript@5.2.2_webpack@5.89.0/node_modules/payload/src/collections/operations/create.ts:285:14)
cms:dev:     at async createHandler (/home/dev/Documents/github/hungvu.tech/node_modules/.pnpm/payload@2.0.15_typescript@5.2.2_webpack@5.89.0/node_modules/payload/src/collections/requestHandlers/create.ts:26:17
```

Here is the sample code to reproduce the error.

```ts
/**
 * @author Hung Vu
 *
 * Observe behavior of server-side HTML serialization.
 */

import { CollectionConfig } from 'payload/types';

import {
  HTMLConverterFeature,
  lexicalEditor,
  lexicalHTML
} from '@payloadcms/richtext-lexical'

const Tests: CollectionConfig = {
  slug: 'tests',
  fields: [
    {
      name: 'fields_richtext',
      // label: 'Collection',
      type: 'richText',
      editor: lexicalEditor({
        features: ({ defaultFeatures }) => [
          ...defaultFeatures,
          // The HTMLConverter Feature is the feature which manages the HTML serializers. If you do not pass any arguments to it, it will use the default serializers.
          HTMLConverterFeature({}),
        ],
      }),
    },
    // Works
    lexicalHTML('fields_richtext', { name: 'fields_richtext_html' }),


    ////////////////////////////////////////
    ////////////////////////////////////////
    //////////////////////////////////////// 
    {
      name: 'groups',
      type: 'group',
      fields: [
        {
          name: 'groups_richtext',
          // label: 'Collection',
          type: 'richText',
          editor: lexicalEditor({
            features: ({ defaultFeatures }) => [
              ...defaultFeatures,
              // The HTMLConverter Feature is the feature which manages the HTML serializers. If you do not pass any arguments to it, it will use the default serializers.
              HTMLConverterFeature({}),
            ],
          }),
        },
        // No error, but empty groups_richtext_html
        // lexicalHTML('groups.groups_richtext', { name: 'groups_richtext_html' }),

        // Error
        // lexicalHTML('groups_richtext', { name: 'groups_richtext_html' }),
      ]
    },
    // Error
    // lexicalHTML('groups_richtext', { name: 'groups_richtext_html' }),
    // lexicalHTML('groups.groups_richtext', { name: 'groups_richtext_html' }),
    
    
    ////////////////////////////////////////
    ////////////////////////////////////////
    //////////////////////////////////////// 
    {
      name: 'arrays',
      type: 'array',
      fields: [
        {
          name: 'arrays_richtext',
          // label: 'Collection',
          type: 'richText',
          editor: lexicalEditor({
            features: ({ defaultFeatures }) => [
              ...defaultFeatures,
              // The HTMLConverter Feature is the feature which manages the HTML serializers. If you do not pass any arguments to it, it will use the default serializers.
              HTMLConverterFeature({}),
            ],
          }),
        },
    // No error, but empty arrays_richtext_html
    // lexicalHTML('arrays[0].arrays_richtext', { name: 'arrays_richtext_html' }),
    
    // Error
    // lexicalHTML('arrays.arrays_richtext', { name: 'arrays_richtext_html' }),
    // lexicalHTML('arrays_richtext', { name: 'arrays_richtext_html' }),
      ]
    },
    // Error
    // lexicalHTML('arrays[0].arrays_richtext', { name: 'arrays_richtext_html' }),
    // lexicalHTML('arrays.arrays_richtext', { name: 'arrays_richtext_html' }),
    // lexicalHTML('arrays_richtext', { name: 'arrays_richtext_html' }),
    

    ////////////////////////////////////////
    ////////////////////////////////////////
    //////////////////////////////////////// 
    {
      label: 'Collapsible',
      type: 'collapsible',
      fields: [
        {
          name: 'collapsible_richtext',
          // label: 'Collection',
          type: 'richText',
          editor: lexicalEditor({
            features: ({ defaultFeatures }) => [
              ...defaultFeatures,
              // The HTMLConverter Feature is the feature which manages the HTML serializers. If you do not pass any arguments to it, it will use the default serializers.
              HTMLConverterFeature({}),
            ],
          }),
        },
        // Error
        // lexicalHTML('collapsible_richtext', { name: 'collapsible_richtext_html' }),
      ]
    },
    // Error
    // lexicalHTML('collapsible_richtext', { name: 'collapsible_richtext_html' }),


    ////////////////////////////////////////
    ////////////////////////////////////////
    //////////////////////////////////////// 
    {
      type: 'row',
      fields: [
        {
          name: 'row_richtext',
          // label: 'Collection',
          type: 'richText',
          editor: lexicalEditor({
            features: ({ defaultFeatures }) => [
              ...defaultFeatures,
              // The HTMLConverter Feature is the feature which manages the HTML serializers. If you do not pass any arguments to it, it will use the default serializers.
              HTMLConverterFeature({}),
            ],
          }),
        },
        // Error
        // lexicalHTML('row_richtext', { name: 'row_richtext_html' }),
      ]
    },
    // Error
    // lexicalHTML('row_richtext', { name: 'row_richtext_html' }),

    ////////////////////////////////////////
    ////////////////////////////////////////
    //////////////////////////////////////// 
    {
      type: 'tabs',
      tabs: [
        {
          label: 'Tab',
          fields: [
            {
              name: 'tab_richtext',
              // label: 'Collection',
              type: 'richText',
              editor: lexicalEditor({
                features: ({ defaultFeatures }) => [
                  ...defaultFeatures,
                  // The HTMLConverter Feature is the feature which manages the HTML serializers. If you do not pass any arguments to it, it will use the default serializers.
                  HTMLConverterFeature({}),
                ],
              }),
            },
            // Error
            // lexicalHTML('tab_richtext', { name: 'tab_richtext_html' }),
          ]
        },
      ]
    },
    // Error
    // lexicalHTML('tab_richtext', { name: 'tab_richtext_html' }),
  ],
};

export default Tests;

```

### Payload Version

2.0.15

### Adapters and Plugins

richtext-lexical v0.1.16

**Comment by Bernstein7z** [comment]
I can reproduce the problem with a similar but another scenario:

```ts
{
      name: "meta",
      label: "Meta",
      type: "group",
      fields: [
        {
          type: "row",
          fields: [
            // some row fields > 4
            {
              name: "description",
              type: "richText",
              label: "Meta Description",
              editor: lexicalEditor({
                features: ({ defaultFeatures }) => [...defaultFeatures, HTMLConverterFeature({})],
              }),
              required: true,
              localized: true,
              admin: {
                width: "100%",
              },
            },
            lexicalHTML("description", { name: "description_html" }),
          ],
        },
       // some group field
      ],
    },
```

The error message:
`Error: Could not find sibling fields of current lexicalHTML field with name description_html`

PayloadCMS version: `2.4.0`

**Comment by AlessioGr** [comment]
> ```ts
> {
>       name: "meta",
>       label: "Meta",
>       type: "group",
>       fields: [
>         {
>           type: "row",
>           fields: [
>             // some row fields > 4
>             {
>               name: "description",
>               type: "richText",
>               label: "Meta Description",
>               editor: lexicalEditor({
>                 features: ({ defaultFeatures }) => [...defaultFeatures, HTMLConverterFeature({})],
>               }),
>               required: true,
>               localized: true,
>               admin: {
>                 width: "100%",
>               },
>             },
>             lexicalHTML("description", { name: "description_html" }),
>           ],
>         },
>        // some group field
>       ],
>     },
> ```

Fixed this in #4440 :)
